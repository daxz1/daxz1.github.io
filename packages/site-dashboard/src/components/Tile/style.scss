@import "../../css/colours";
@import "../../css/font";

.tile {
  font-family: 'Raleway', sans-serif;
  border: 1px solid $shark;
  background: $shark;
  padding: 10px;
  border-radius: 3px;
  color: $white;

  // Widget Style 1
  .tress {
    &__header {
      border-bottom: 1px solid rgba(255,255,255, 0.2);
      font-size: 12px;
      font-weight: 200;
      line-height: 24px;
      text-align: center;
    }
    &__title {
      font-size: 16px;
      font-weight: 200;
      margin: 20px 0 20px 0;
      text-align: center;
    }
    &__content {
      text-align: center;
      font-size: 32px;
      padding: 30px;
      &--sushi {
        background: $sushi;
      }
      &--flame {
        background: $flame;
      }
      &--westside {
        background: $westside;
      }
      &--java {
        background: $java;
      }
    }
  }
}

.loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #3498db;
  width: 80px;
  height: 80px;
  -webkit-animation: spin 2s linear infinite; /* Safari */
  animation: spin 2s linear infinite;
  margin: 0 auto;
}

/* Safari */
@-webkit-keyframes spin {
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}