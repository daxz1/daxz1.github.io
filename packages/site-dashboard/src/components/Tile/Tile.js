import React, { memo } from "react";
import "./style.scss";

/**
 *
 * @param header
 * @param title
 * @param content
 * @param cssClasses
 * @returns {JSX.Element}
 * @constructor
 */
export const Tile = memo(({header, title, content, cssClasses}) => {
  return (
    <div className="tile fade__in">
      <>
        <div className="tress__header">
          {header}
        </div>
        <div className="tress__title">
          {title}
        </div>
        <div className={['tress__content', cssClasses].join(' ')}>
          {content}
        </div>
      </>
    </div>
  );
});