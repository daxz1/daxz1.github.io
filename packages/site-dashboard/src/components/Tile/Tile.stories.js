import React from "react";
import { Tile } from "./Tile";

export default {
  title: 'components/Tile',
  component: Tile,
  argTypes: {
    header: { control: 'text' },
    title: { control: 'text' },
    content: { control: 'text' },
    cssClasses: {
      control: {
        type: 'select',
        options: ['tress__content--sushi', 'tress__content--flame', 'tress__content--westside', 'tress__content--java']
      },
    },
  },
}

const Template = (args) => <Tile {...args} />;

export const TileWithContent = Template.bind({});

TileWithContent.args = {
  header: 'Header Text',
  title: 'Title Text',
  content: 'Content Text',
  cssClasses: 'tress__content--sushi'
}