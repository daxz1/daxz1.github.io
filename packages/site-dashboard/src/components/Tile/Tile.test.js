/*global describe, test */

import React from "react";
import { create } from "react-test-renderer";
import { Tile } from "./Tile";

describe("Tile Component", () => {
  test("Matches the snapshot", () => {
    const Application = create(
      <Tile
        header="Header"
        title="Title"
        content="Content"
        cssClasses={["tress__content--sushi"]}
      />
    );
    expect(Application.toJSON()).toMatchSnapshot();
  });
});
