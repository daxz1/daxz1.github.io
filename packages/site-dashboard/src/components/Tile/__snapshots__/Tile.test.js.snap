// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Component Matches the snapshot 1`] = `
<div
  className="tile fade__in"
>
  <div
    className="tress__header"
  >
    Header
  </div>
  <div
    className="tress__title"
  >
    Title
  </div>
  <div
    className="tress__content tress__content--sushi"
  >
    Content
  </div>
</div>
`;
